src/allmydata/storage_client.py

```diff
@@ -1189,3 +1190,53 @@ def advise_corrupt_share(
             )
         else:
             raise NotImplementedError()  # future tickets
+
+    @defer.inlineCallbacks
+    def slot_readv(self, storage_index, shares, readv):
+        mutable_client = StorageClientMutables(self._http_client)
+        reads = {}
+        # TODO if shares list is empty, that means list all shares, so we need
+        # to do a query to get that.
+        assert shares  # TODO replace with call to list shares
+        for share_number in shares:
+            share_reads = reads[share_number] = []
+            for (offset, length) in readv:
+                r = yield mutable_client.read_share_chunk(
+                    storage_index, share_number, offset, length
+                )
```

Comment:
maybe worth noting this as a good potential spot for optimization in the future - to make at least some of these requests in parallel?

Reply:
I just made it parallel immediately.